src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -488,7 +488,7 @@ private void OnStateChanged(object sender, RachisConsensus.StateTransition state
                 {
                     RefreshOutgoingTasks();
                 }
-                if (state.To == RachisConsensus.State.LeaderElect)
+                if (state.To == RachisConsensus.State.LeaderElect || state.To == RachisConsensus.State.Leader)
                 {
                     _engine.CurrentLeader.OnNodeStatusChange += OnTopologyChanged;
```

Comment:
There is a problem, if I'm first leader elect, and I register, then leader, I double register.